LDAR_Sim/model_code/methods/base_crew.py

```diff
@@ -0,0 +1,402 @@
+                                facility_ID = site['facility_ID']
+                                found_site = True
+                                travel_time = int(np.random.choice(self.config['t_bw_sites']))
+                                # Update site
+                                site['{}_surveys_conducted'.format(m_name)] += 1
+                                site['{}_surveys_done_this_year'.format(m_name)] += 1
+                                site['{}_t_since_last_LDAR'.format(m_name)] = 0
+                                break
+
+                        else:
+                            site['{}_attempted_today?'.format(m_name)] = True
+
+        if self.scheduling['route_planning']:
+            if len(Site_T) > 0:
+                j = Site_T.index(min(Site_T))
+                site = s_list[j]
+                facility_ID = site['facility_ID']
+                travel_time = min(Site_T)*60
+                found_site = True
+
+                # Update site
+                site['{}_surveys_conducted'.format(m_name)] += 1
+                site['{}_surveys_done_this_year'.format(m_name)] += 1
+                site['{}_t_since_last_LDAR'.format(m_name)] = 0
+
+        return (facility_ID, found_site, site, travel_time)
+
+    def visit_site(self, site, travel_time=None):
+        """
+        Look for emissions at the chosen site.
+        """
+        m_name = self.config['name']
+
+        # Aggregate true emissions to equipment and site level; get list of leaks present
+        leaks_present, equipment_rates, site_true_rate = aggregate(site, self.state['leaks'])
+
+        # Add vented emissions
+        venting = 0
+        if self.parameters['consider_venting']:
+            venting = self.state['empirical_vents'][
+                np.random.randint(0, len(self.state['empirical_vents']))]
+        site_true_rate += venting
+        for rate in range(len(equipment_rates)):
+            equipment_rates[rate] += venting/int(site['equipment_groups'])
+
+        # Placeholder for is leaks detected?, could be used to split up reporting
+        # from detect function in the future
+
+        _ = self.detect_leaks(
+            site, leaks_present, equipment_rates, site_true_rate, venting)
+
+        # --- Update time ---
+        self.state['t'].current_date += timedelta(minutes=int(site['{}_time'.format(m_name)]))
+        # if travel time was not generated when choosing site
+        if not travel_time:
+            travel_time = np.random.choice(self.config['t_bw_sites'])
+        self.state['t'].current_date += timedelta(
+            minutes=int(travel_time))
+        self.timeseries['{}_sites_visited'.format(m_name)][self.state['t'].current_timestep] += 1
+
+        if self.config['is_follow_up']:
+            # Remove site from flag pool
+            site['currently_flagged'] = False
+        return
+
+    def detect_leaks(self, site, leaks_present, equipment_rates, site_true_rate, venting):
```

Comment:
Should this be "detect_emissions"? Also, where is this used? I thought this functionality was implemented already for aircraft... apologies if I don't understand.

Reply:
Copied from OGI_Camera class. Changed to detect_leaks